airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -828,12 +849,15 @@ def _submit_job(
         """Submit a new job and get the job id for polling the status using Trigger."""
         configuration = {"query": {"query": self.sql, "useLegacySql": self.use_legacy_sql}}
 
+        if self.encryption_configuration is not None:
+            configuration["query"]["destinationEncryptionConfiguration"] = self.encryption_configuration
+
         return hook.insert_job(
             configuration=configuration,
             project_id=hook.project_id,
             location=self.location,
             job_id=job_id,
-            nowait=False,
+            nowait=True,
```

Comment:
Could you please explain what's the reason for this change?

Reply:
At the beginning I was getting an error for this operator after digging, I found out when I changed this parameter to True, everything started to work fine. Here is the task run log in case we have `nowait=False`
```
[2024-04-26, 15:52:54 CEST] {taskinstance.py:2972} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 477, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/airflow/airflow/models/taskinstance.py", line 440, in _execute_callable
    return ExecutionCallableRunner(
  File "/opt/airflow/airflow/utils/operator_helpers.py", line 250, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/airflow/models/baseoperator.py", line 405, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/operators/bigquery.py", line 866, in execute
    job = self._submit_job(hook, job_id="")
  File "/opt/airflow/airflow/providers/google/cloud/operators/bigquery.py", line 855, in _submit_job
    return hook.insert_job(
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 524, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/bigquery.py", line 1674, in insert_job
    job_api_repr.result(timeout=timeout, retry=retry)
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/query.py", line 1626, in result
    while not is_job_done():
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/query.py", line 1551, in is_job_done
    if self.done(retry=retry, timeout=timeout):
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 938, in done
    self.reload(retry=retry, timeout=timeout)
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 828, in reload
    api_response = client._call_api(
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/client.py", line 831, in _call_api
    return call()
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 293, in retry_wrapped_func
    return retry_target(
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 153, in retry_target
    _retry_error_helper(
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_base.py", line 212, in _retry_error_helper
    raise final_exc from source_exc
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 144, in retry_target
    result = target()
  File "/usr/local/lib/python3.8/site-packages/google/cloud/_http/__init__.py", line 494, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.NotFound: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/airflow-system-tests-cmek/jobs/airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739?prettyPrint=false: Not found: Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
Location: None
Job ID: airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
 [{'@type': 'type.googleapis.com/google.rpc.DebugInfo', 'detail': '[NOT_FOUND] message=Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739 not found debug=time: 2024-04-26T06:52:54.171-07:00 errorProto=code: "NOT_FOUND"\nargument: "Job"\nargument: "airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739"\n\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:347)\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:351)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.lambda$getInternal$23(HelixJobRosy.java:2749)\n\tat java.base/java.util.Optional.orElseThrow(Unknown Source)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.getInternal(HelixJobRosy.java:2749)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.get(HelixJobRosy.java:3797)\n\tat jdk.internal.reflect.GeneratedMethodAccessor294.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)\n\tat java.base/java.lang.reflect.Method.invoke(Unknown Source)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$innerContinuation$3(RpcRequestProxy.java:435)\n\tat com.google.cloud.helix.common.rosy.RosyRequestDapperHookFactory$TracingRequestHook.call(RosyRequestDapperHookFactory.java:88)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestCredsHookFactory$1.call(RosyRequestCredsHookFactory.java:56)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestConcurrentCallsHookFactory$Hook.call(RosyRequestConcurrentCallsHookFactory.java:101)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestVarzHookFactory$Hook.call(RosyRequestVarzHookFactory.java:464)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestAuditHookFactory$1.call(RosyRequestAuditHookFactory.java:107)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RequestSecurityExtensionForGwsHookFactory$1.call(RequestSecurityExtensionForGwsHookFactory.java:69)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestSecurityContextHookFactory$1.call(RosyRequestSecurityContextHookFactory.java:83)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestContextHookFactory.call(RosyRequestContextHookFactory.java:60)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.invoke(RpcRequestProxy.java:666)\n\tat com.sun.proxy.$Proxy53.get(Unknown Source)\n\tat com.google.cloud.helix.proto.proto2api.HelixJobService$ServiceParameters$2.handleRequest(HelixJobService.java:925)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$RpcApplicationHandlerAdaptor.handleRequest(RpcServerInterceptor2Util.java:82)\n\tat com.google.net.rpc3.impl.server.AggregatedRpcServerInterceptors.interceptRpc(AggregatedRpcServerInterceptors.java:100)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$InterceptedApplicationHandlerImpl.handleRequest(RpcServerInterceptor2Util.java:67)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplicationWithCancellation(RpcServerInternalContext.java:693)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.lambda$runRpcInApplication$0(RpcServerInternalContext.java:658)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplication(RpcServerInternalContext.java:658)\n\tat com.google.net.rpc3.util.RpcInProcessConnector$ServerInternalContext.lambda$runWithExecutor$1(RpcInProcessConnector.java:1973)\n\tat com.google.common.context.ContextRunnable.runInContext(ContextRunnable.java:83)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.tracing.GenericContextCallback.runInInheritedContext(GenericContextCallback.java:75)\n\tat com.google.common.context.ContextRunnable.run(ContextRunnable.java:74)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n'}]
[2024-04-26, 15:52:54 CEST] {taskinstance.py:1259} INFO - Marking task as FAILED. dag_id=bigquery_queries, task_id=table_check, execution_date=20240426T135248, start_date=20240426T135253, end_date=20240426T135254
[2024-04-26, 15:52:54 CEST] {standard_task_runner.py:112} ERROR - Failed to execute job 6 for task table_check (404 GET https://bigquery.googleapis.com/bigquery/v2/projects/airflow-system-tests-cmek/jobs/airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739?prettyPrint=false: Not found: Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
Location: None
Job ID: airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
 [{'@type': 'type.googleapis.com/google.rpc.DebugInfo', 'detail': '[NOT_FOUND] message=Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739 not found debug=time: 2024-04-26T06:52:54.171-07:00 errorProto=code: "NOT_FOUND"\nargument: "Job"\nargument: "airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739"\n\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:347)\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:351)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.lambda$getInternal$23(HelixJobRosy.java:2749)\n\tat java.base/java.util.Optional.orElseThrow(Unknown Source)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.getInternal(HelixJobRosy.java:2749)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.get(HelixJobRosy.java:3797)\n\tat jdk.internal.reflect.GeneratedMethodAccessor294.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)\n\tat java.base/java.lang.reflect.Method.invoke(Unknown Source)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$innerContinuation$3(RpcRequestProxy.java:435)\n\tat com.google.cloud.helix.common.rosy.RosyRequestDapperHookFactory$TracingRequestHook.call(RosyRequestDapperHookFactory.java:88)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestCredsHookFactory$1.call(RosyRequestCredsHookFactory.java:56)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestConcurrentCallsHookFactory$Hook.call(RosyRequestConcurrentCallsHookFactory.java:101)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestVarzHookFactory$Hook.call(RosyRequestVarzHookFactory.java:464)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestAuditHookFactory$1.call(RosyRequestAuditHookFactory.java:107)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RequestSecurityExtensionForGwsHookFactory$1.call(RequestSecurityExtensionForGwsHookFactory.java:69)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestSecurityContextHookFactory$1.call(RosyRequestSecurityContextHookFactory.java:83)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestContextHookFactory.call(RosyRequestContextHookFactory.java:60)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.invoke(RpcRequestProxy.java:666)\n\tat com.sun.proxy.$Proxy53.get(Unknown Source)\n\tat com.google.cloud.helix.proto.proto2api.HelixJobService$ServiceParameters$2.handleRequest(HelixJobService.java:925)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$RpcApplicationHandlerAdaptor.handleRequest(RpcServerInterceptor2Util.java:82)\n\tat com.google.net.rpc3.impl.server.AggregatedRpcServerInterceptors.interceptRpc(AggregatedRpcServerInterceptors.java:100)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$InterceptedApplicationHandlerImpl.handleRequest(RpcServerInterceptor2Util.java:67)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplicationWithCancellation(RpcServerInternalContext.java:693)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.lambda$runRpcInApplication$0(RpcServerInternalContext.java:658)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplication(RpcServerInternalContext.java:658)\n\tat com.google.net.rpc3.util.RpcInProcessConnector$ServerInternalContext.lambda$runWithExecutor$1(RpcInProcessConnector.java:1973)\n\tat com.google.common.context.ContextRunnable.runInContext(ContextRunnable.java:83)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.tracing.GenericContextCallback.runInInheritedContext(GenericContextCallback.java:75)\n\tat com.google.common.context.ContextRunnable.run(ContextRunnable.java:74)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n'}]; 3485)
Traceback (most recent call last):
  File "/opt/airflow/airflow/task/task_runner/standard_task_runner.py", line 105, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/opt/airflow/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/opt/airflow/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/opt/airflow/airflow/cli/commands/task_command.py", line 476, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/opt/airflow/airflow/cli/commands/task_command.py", line 253, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/opt/airflow/airflow/cli/commands/task_command.py", line 335, in _run_raw_task
    return ti._run_raw_task(
  File "/opt/airflow/airflow/utils/session.py", line 84, in wrapper
    return func(*args, session=session, **kwargs)
  File "/opt/airflow/airflow/models/taskinstance.py", line 2548, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/opt/airflow/airflow/models/taskinstance.py", line 2744, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/opt/airflow/airflow/models/taskinstance.py", line 2768, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/opt/airflow/airflow/models/taskinstance.py", line 477, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/airflow/airflow/models/taskinstance.py", line 440, in _execute_callable
    return ExecutionCallableRunner(
  File "/opt/airflow/airflow/utils/operator_helpers.py", line 250, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/airflow/models/baseoperator.py", line 405, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/operators/bigquery.py", line 866, in execute
    job = self._submit_job(hook, job_id="")
  File "/opt/airflow/airflow/providers/google/cloud/operators/bigquery.py", line 855, in _submit_job
    return hook.insert_job(
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 524, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/bigquery.py", line 1674, in insert_job
    job_api_repr.result(timeout=timeout, retry=retry)
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/query.py", line 1626, in result
    while not is_job_done():
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/query.py", line 1551, in is_job_done
    if self.done(retry=retry, timeout=timeout):
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 938, in done
    self.reload(retry=retry, timeout=timeout)
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 828, in reload
    api_response = client._call_api(
  File "/usr/local/lib/python3.8/site-packages/google/cloud/bigquery/client.py", line 831, in _call_api
    return call()
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 293, in retry_wrapped_func
    return retry_target(
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 153, in retry_target
    _retry_error_helper(
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_base.py", line 212, in _retry_error_helper
    raise final_exc from source_exc
  File "/usr/local/lib/python3.8/site-packages/google/api_core/retry/retry_unary.py", line 144, in retry_target
    result = target()
  File "/usr/local/lib/python3.8/site-packages/google/cloud/_http/__init__.py", line 494, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.NotFound: 404 GET https://bigquery.googleapis.com/bigquery/v2/projects/airflow-system-tests-cmek/jobs/airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739?prettyPrint=false: Not found: Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
Location: None
Job ID: airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739
 [{'@type': 'type.googleapis.com/google.rpc.DebugInfo', 'detail': '[NOT_FOUND] message=Job airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739 not found debug=time: 2024-04-26T06:52:54.171-07:00 errorProto=code: "NOT_FOUND"\nargument: "Job"\nargument: "airflow-system-tests-cmek:airflow_1714139573775519_d1b7d4d581f0a75dbb40f0b4f7e69739"\n\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:347)\n\tat com.google.cloud.helix.common.Exceptions$Public.resourceNotFound(Exceptions.java:351)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.lambda$getInternal$23(HelixJobRosy.java:2749)\n\tat java.base/java.util.Optional.orElseThrow(Unknown Source)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.getInternal(HelixJobRosy.java:2749)\n\tat com.google.cloud.helix.server.job.rosy.HelixJobRosy.get(HelixJobRosy.java:3797)\n\tat jdk.internal.reflect.GeneratedMethodAccessor294.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)\n\tat java.base/java.lang.reflect.Method.invoke(Unknown Source)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$innerContinuation$3(RpcRequestProxy.java:435)\n\tat com.google.cloud.helix.common.rosy.RosyRequestDapperHookFactory$TracingRequestHook.call(RosyRequestDapperHookFactory.java:88)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestCredsHookFactory$1.call(RosyRequestCredsHookFactory.java:56)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestConcurrentCallsHookFactory$Hook.call(RosyRequestConcurrentCallsHookFactory.java:101)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestVarzHookFactory$Hook.call(RosyRequestVarzHookFactory.java:464)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestAuditHookFactory$1.call(RosyRequestAuditHookFactory.java:107)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RequestSecurityExtensionForGwsHookFactory$1.call(RequestSecurityExtensionForGwsHookFactory.java:69)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestSecurityContextHookFactory$1.call(RosyRequestSecurityContextHookFactory.java:83)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RosyRequestContextHookFactory.call(RosyRequestContextHookFactory.java:60)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.lambda$makeContinuation$4(RpcRequestProxy.java:461)\n\tat com.google.cloud.helix.common.rosy.RpcRequestProxy.invoke(RpcRequestProxy.java:666)\n\tat com.sun.proxy.$Proxy53.get(Unknown Source)\n\tat com.google.cloud.helix.proto.proto2api.HelixJobService$ServiceParameters$2.handleRequest(HelixJobService.java:925)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$RpcApplicationHandlerAdaptor.handleRequest(RpcServerInterceptor2Util.java:82)\n\tat com.google.net.rpc3.impl.server.AggregatedRpcServerInterceptors.interceptRpc(AggregatedRpcServerInterceptors.java:100)\n\tat com.google.net.rpc3.impl.server.RpcServerInterceptor2Util$InterceptedApplicationHandlerImpl.handleRequest(RpcServerInterceptor2Util.java:67)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplicationWithCancellation(RpcServerInternalContext.java:693)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.lambda$runRpcInApplication$0(RpcServerInternalContext.java:658)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.net.rpc3.impl.server.RpcServerInternalContext.runRpcInApplication(RpcServerInternalContext.java:658)\n\tat com.google.net.rpc3.util.RpcInProcessConnector$ServerInternalContext.lambda$runWithExecutor$1(RpcInProcessConnector.java:1973)\n\tat com.google.common.context.ContextRunnable.runInContext(ContextRunnable.java:83)\n\tat io.grpc.Context.run(Context.java:536)\n\tat com.google.tracing.GenericContextCallback.runInInheritedContext(GenericContextCallback.java:75)\n\tat com.google.common.context.ContextRunnable.run(ContextRunnable.java:74)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n'}]
[2024-04-26, 15:52:54 CEST] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-26, 15:52:54 CEST] {taskinstance.py:3571} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26, 15:52:54 CEST] {local_task_job_runner.py:222} ▲▲▲ Log group end
```
I suspected that maybe we need to wait before checking the query result, so I did not revert this change. If you have any other ideas please feel free

Reply:
I managed to reproduce this error and it seems to me that it's not related with CMEK, and thus I don't think we should provide this change here.
Probably this error is caused by another bug that we have in our backlog, and we are planning to solve it asap.